joern-cli/frontends/c2cpg/src/main/scala/io/joern/c2cpg/astcreation/AstForPrimitivesCreator.scala

```diff
@@ -49,7 +50,11 @@ trait AstForPrimitivesCreator { this: AstCreator =>
           case Some(binding: (CPPVariable | CVariable)) =>
             Try(binding.getScope).toOption
               .collect { case n: IASTInternalScope => n.getPhysicalNode }
-              .exists(_.isInstanceOf[IASTTranslationUnit])
+              .exists {
+                case _: IASTTranslationUnit        => true
+                case n: ICPPASTNamespaceDefinition => n.getParent.isInstanceOf[IASTTranslationUnit]
```

Comment:
Stuff directly under file-global namespaces is also global.

Reply:
do we explicitly need to handle nested namespaces here, or does CDT collapse those?

```c++
namespace N { namespace M { const int global = 2; }; };
```